File: src/main/java/cz/grunwjir/csasdemo/gmaps/client/GMapsClient.java
package cz.grunwjir.csasdemo.gmaps.client;

/**
 * The client for Google Maps API.
 * Now supports only one endpoint to search places.
 *
 * @author Jiri Grunwald
 */
public interface GMapsClient {

    /**
     * Search place by the given text.
     *
     * @param placeName place name, can also contain address (if place name is ambiguous)
     * @return found place that most meets the criteria, otherwise null
     */
    PlaceData findPlaceFromText(String placeName);
}

File: src/main/kotlin/cz/grunwjir/csasdemo/gmaps/client/GMapsClient.kt
package cz.grunwjir.csasdemo.gmaps.client

/**
 * The client for Google Maps API.
 * Now supports only one endpoint to search places.
 *
 * @author Jiri Grunwald
 */
interface GMapsClient {
    /**
     * Search place by the given text.
     *
     * @param placeName place name, can also contain address (if place name is ambiguous)
     * @return found place that most meets the criteria, otherwise null
     */
    fun findPlaceFromText(placeName: String): PlaceData?
}